dataactbroker/handlers/fileHandler.py

```diff
@@ -479,9 +498,134 @@ def checkD1File(self, submission_id=None):
+        else:
+            validationJob = None
+        responseDict = {}
+        responseDict["status"] = self.mapGenerateStatus(uploadJob, validationJob)
+        responseDict["file_type"] = file_type
+        responseDict["message"] = uploadJob.error_message or ""
+        if uploadJob.filename is None:
+            responseDict["url"] = ""
+        else:
+            if CONFIG_BROKER["use_aws"]:
+                responseDict["url"] = s3UrlHandler().getSignedUrl("d-files",uploadJob.filename, bucketRoute=None, method="GET")
+            else:
+                responseDict["url"] = uploadJob.filename
+
+        # Pull start and end from jobs table if D1 or D2
+        if file_type in ["D1","D2"]:
+            responseDict["start"] = uploadJob.start_date.strftime("%m/%d/%Y") if uploadJob.start_date is not None else ""
+            responseDict["end"] = uploadJob.end_date.strftime("%m/%d/%Y") if uploadJob.end_date is not None else ""
+
+        return JsonResponse.create(StatusCode.OK,responseDict)
+
+    def mapGenerateStatus(self, uploadJob, validationJob = None):
+        """ Maps job status to file generation statuses expected by frontend """
+        uploadStatus = self.interfaces.jobDb.getJobStatusNameById(uploadJob.job_status_id)
+        if validationJob is None:
+            errorsPresent = False
+        else:
+            validationStatus = self.interfaces.jobDb.getJobStatusNameById(validationJob.job_status_id)
+            if self.interfaces.errorDb.checkNumberOfErrorsByJobId(validationJob.job_id, self.interfaces.validationDb, errorType = "fatal") > 0:
+                errorsPresent = True
+            else:
+                errorsPresent = False
+
+        if uploadStatus in ["waiting", "ready"]:
+            result =  "invalid"
+        elif uploadStatus == "running":
+            result =  "waiting"
+        elif uploadStatus in ["invalid", "failed"]:
+            if uploadJob.error_message is None:
+                uploadJob.error_message = "Upload job failed without error message"
+            result = "failed"
+        elif uploadStatus == "finished":
+            if validationJob is None:
+                result = "finished"
+            else:
+                # If validation job exists, check that status as well
+                if validationStatus in ["waiting", "ready", "running"]:
+                    result = "waiting"
+                elif validationStatus in ["invalid", "failed"]:
+                    if validationJob.error_message is None:
+                        if uploadJob.error_message is None:
+                            if validationStatus == "invalid":
+                                uploadJob.error_message = "Generated file had file-level errors"
+                            else:
+                                uploadJob.error_message = "Validation job had an internal error"
```

Comment:
The logic here is nested _very_ deeply (we're at 6 levels of `if`). Is there a way to simplify the logic so that it's easier to follow? Right now I need to effectively execute the function in my head to figure out which path is taken.